src/main/java/spoon/reflect/factory/CodeFactory.java

```diff
@@ -139,7 +139,7 @@ public CodeFactory(Factory factory) {
 		fieldReference.setDeclaringType(type);
 
 		CtFieldRead<Class<T>> fieldRead = factory.Core().createFieldRead();
-		fieldRead.setType(classType);
+		fieldRead.setType(factory.Core().clone(classType));
```

Comment:
Do you have an use case where the type of the field read isn't equals of the field reference?